src/Cli/dotnet/Commands/Package/Add/PackageAddCommand.cs

```diff
@@ -133,4 +139,214 @@ private string[] TransformArgs(PackageIdentityWithRange packageId, string tempDg
+
+        if (!_parseResult.GetValue(PackageAddCommandParser.NoRestoreOption))
+        {
+            // Restore.
+            int exitCode = command.Execute();
+            if (exitCode != 0)
+            {
+                // If restore fails, revert any changes made.
+                central?.Revert();
+                return exitCode;
+            }
+
+            // If no version was specified by the user, save the actually restored version.
+            if (!hasVersion && !skipUpdate)
+            {
+                var projectAssetsFile = projectInstance.GetProperty("ProjectAssetsFile")?.EvaluatedValue;
+                if (!File.Exists(projectAssetsFile))
+                {
+                    Reporter.Verbose.WriteLine($"Assets file does not exist: {projectAssetsFile}");
+                }
+                else
+                {
+                    var lockFile = new LockFileFormat().Read(projectAssetsFile);
+                    var library = lockFile.Libraries.FirstOrDefault(l => string.Equals(l.Name, _packageId.Id, StringComparison.OrdinalIgnoreCase));
+                    if (library != null)
+                    {
+                        var restoredVersion = library.Version.ToString();
+                        if (central is { } centralValue)
+                        {
+                            centralValue.Update(restoredVersion);
+                            local.Save();
+                        }
+                        else
+                        {
+                            local.Update(restoredVersion);
+                        }
+
+                        return 0;
+                    }
+                }
+            }
+        }
+
+        central?.Save();
+        local.Save();
+        return 0;
+
+        (Action Save, Action<string> Update) SetLocalVersion(string? version)
+        {
+            // Add #:package directive to the C# file.
+            var file = SourceFile.Load(fullPath);
+            var editor = FileBasedAppSourceEditor.Load(file);
+            editor.Add(new CSharpDirective.Package(default) { Name = _packageId.Id, Version = version });
+            command.Directives = editor.Directives;
+            return (Save, Update);
+
+            void Save()
+            {
+                editor.SourceFile.Save();
+            }
+
+            void Update(string value)
+            {
+                // Update the C# file with the given version.
+                editor.Add(new CSharpDirective.Package(default) { Name = _packageId.Id, Version = value });
+                editor.SourceFile.Save();
+            }
+        }
+
+        (Action Revert, Action<string> Update, Action Save)? SetCentralVersion(string version)
+        {
+            // Find out whether CPM is enabled.
+            if (!string.Equals(projectInstance.GetProperty("ManagePackageVersionsCentrally")?.EvaluatedValue, bool.TrueString, StringComparison.OrdinalIgnoreCase))
```

Comment:
There are a lot of values in MSBuild that are _true_. For example `1` is considered _true_. Does `EvaluatedValue` normalize `bool` values to the more canonical `True / False`?

Reply:
No, I need to handle those myself, thanks.